float64 base_angle
float64 lower_angle
float64 upper_angle
